packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -517,19 +518,46 @@ export async function resolveMetadataItems({
   return metadataItems
 }
 
+type WithTitle = { title?: AbsoluteTemplateString | null }
+type WithDescription = { description?: string | null }
+
+const hasTitle = (metadata: WithTitle | null) => !!metadata?.title?.absolute
+
+function inheritFromMetadata(
+  metadata: ResolvedMetadata,
+  target: (WithTitle & WithDescription) | null
+) {
+  if (target) {
+    if (!hasTitle(target) && metadata.title && hasTitle(metadata)) {
+      target.title = metadata.title
+    }
```

Comment:
would there ever be a case where someone explicitly doesn't want to set a og/twitter title/description, but can no longer do that because it's forced to inherit? 

basically wondering if we should treat `undefined` differently than `null`, but maybe that doesn't make sense. What do you think?